src/Terrain/terrain.cpp

```diff
@@ -995,6 +968,7 @@ int Terrain::qb_read(const char * path, const std::map<uint32_t, std::pair<const
     }
     fclose(file);
     std::cout << "    tiles read: " << tiles_read << std::endl;
+    free(name);
```

Comment:
NO NO NO NO NO

NEVER NEVER use malloc and free in C++

USE NEW AND DELETE
OR JUST DO THE RIGHT THING AND USE `std::string` WHICH DOES THE RAII FOR YOU